elasticsearch/build-contents-manifest.sh

```diff
@@ -0,0 +1,204 @@
+    echo $1 | sed -n '/^\(..*\)-\('"$ES_VER"'[.].*redhat-[1-9][0-9]*\)[.]jar/{s/^\(..*\)-\('"$ES_VER"'[.].*redhat-[1-9][0-9]*\)[.]jar/\1/;p}'
+}
+
+get_redhat_jar_ver() {
+    # e.g. elasticsearch-prometheus-exporter-5.6.13.1-redhat-1.jar
+    echo $1 | sed -n '/^\(..*\)-\('"$ES_VER"'[.].*redhat-[1-9][0-9]*\)[.]jar/{s/^\(..*\)-\('"$ES_VER"'[.].*redhat-[1-9][0-9]*\)[.]jar/\2/;p}'
+}
+
+# see if this looks like a valid version number
+# for now we require it to start with a digit, because
+# have to handle valid version numbers like
+# 5.6.13.redhat-1
+# 3.10.6.Final
+is_valid_ver() {
+    echo "$1" | grep -q '^[0-9]'
+}
+
+get_version_from_jar() {
+    local jarfile=$1
+    local ver=""
+    local metafile
+    local file
+    local ii
+    local tmpjardir=$( mktemp -d ${TMPDIR:-/tmp}/es-manifest-XXXXXXXX )
+    trap "rm -rf $tmpjardir" RETURN
+    if ! unzip -qq -d $tmpjardir $jarfile ; then
+        echo ERROR: could not unzip $jarfile
+        exit 1
+    fi
+    metafile=$tmpjardir/META-INF/MANIFEST.MF
+    if [ -f $metafile ] ; then
+        # looks like Implementation-Version: is the most reliable
+        # indicator of the version we need for tracking
+        # except in some cases where it is lying . . .
+        case "$jarfile" in
+        */jcodings-*) ver="" ;;
+        */joni-*) ver="" ;;
+        *) ver=$( awk '/^Implementation-Version:/ {gsub(/\r/,""); print $2}' $metafile ) ;;
+        esac
+    fi
+    if ! is_valid_ver "$ver" && [ -d $tmpjardir/META-INF/maven ] && \
+        metafile=$( find $tmpjardir/META-INF/maven -name pom.properties ) && \
+        [ -n "$metafile" ] ; then
+        ii=0
+        for file in $metafile ; do
+            ii=$(( ++ii ))
+        done
+        if [ $ii -eq 1 ] ; then
+            ver=$( awk -F'[ =]+' '/^version=/ {gsub(/\r/,""); print $2}' $metafile )
+        fi
+    fi
+    if is_valid_ver "$ver" ; then
+        echo "$ver"
+        verbose DEBUG: get_version_from_jar jarfile $jarfile version $ver
+        return 0
+    fi
+    verbose DEBUG: get_version_from_jar jarfile $jarfile could not get version
+    return 1
+}
+
+get_package_for_jar() {
+    local jarname=$1
+    local sha=$2
+    local jarver=${3:-}
+    local guess_basename=$( get_redhat_jar_basename $jarname )
+    if [ -z "$guess_basename" ] ; then
+        guess_basename=$( get_compound_jar_basename $jarname )
+    fi
+    if [ -z "$guess_basename" ] ; then
+        guess_basename=$( get_jar_basename $jarname )
+    fi
+    local ver=$( get_redhat_jar_ver $jarname )
+    if [ -z "$ver" ] ; then
+        ver=$( get_compound_jar_ver $jarname )
+    fi
+    if [ -z "$ver" ] ; then
+        ver=$( get_jar_ver $jarname )
+    fi
+    if [ -n "${jarver:-}" -a ! "${jarver:-}" = "${ver}" ] ; then
+        verbose ERROR: $jarname version $ver does not match version from jar $jarver
+    fi
+    local basename
+    case $jarname in
+    accessors-smart-*) basename=net.minidev-accessors-smart ;;
```

Comment:
Speculating with no understanding if it is possible, but is this retrievable from the maven dependency goal?  Seems like we could miss something if we dont remember to update this list

Reply:
Just ignore this for now.  There should be a better way to extract Java information from the middleware team.